[kodi] Switched to shared instance of WebSocketClient

cweitkamp opened the pull request:
- Switched to shared instance of WebSocketClient using WebSocketFactory
- Minor code cleanups to fix sat findings

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

cweitkamp commented:
@kaikreuzer May I ask you to habe a look if this implementation is okay? Thanks.

cweitkamp commented on addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java:
```diff
@@ -74,9 +74,13 @@ public synchronized void open() throws Exception {
-        if (!client.isStarted()) {
-            client.start();
-        }
+        // if (!client.isStarted()) {
```
Not need, right?

martinvw replied:
Indeed, you can trust the framework to start the common client for you

cweitkamp commented on addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java:
```diff
@@ -96,18 +100,17 @@ public void close() {
-        } catch (Exception e) {
-            logger.debug("Exception during closing the websocket: {}", e.getMessage(), e);
-        }
+        // try {
```
Not need, right?

martinvw replied:
👍

cweitkamp commented:
> Did you also test it

Yes, of course. But using a single Kodi instance. Will do a second test with two one more different connections.

Unfortunately not today. Maybe tomorrow.

martinvw commented:
> Unfortunately not today. Maybe tomorrow.

I trust it enough looking at the diff, so when the comments are gone feel free to merge 👍

cweitkamp commented:
@martinvw Thanks for your trust and your review. I removed the comments and changed the exception type for one method to a more specific one. Merge is up to you.

martinvw commented:
Thanks for both your PR's
